### PR TITLE
Store locale in localstorage

### LINKED ISSUE #27: Write locale to and read locale from browser storage
Currently locale change is restricted to current page. As soon as page reloads, locale reverts to the default, which is 'en'.

**Describe the steps to reproduce the bug**
1. Open an page powered by Vocabulary.
1. Change locale.
1. Navigate to a different page.

The locale is supposed to remain the chosen one. But it reverts to the default one. This is because the locale is never given any form of permanence. The goal is to write the chosen locale to the local storage and read from there on every reload. This will allow the locale change to persist across sessions and reloads.

### PR DESCRIPTION
**Pull request by Pilgrim001**
Fixes #27

**Description**
Added logic to store selected locale in browser local storage so as to prevent the locale from reverting to the default whenever the user refreshes the page or navigates to a new route.

**Other information**

**Checklist:**
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by dhruvkb** [comment]
This is working as expected. We will have to wait as the relevant issue #27 was "in-progress" before this PR.

**Comment by Pilgrim001** [comment]
> Can the changes made to `src/i18n.js` be moved into `src/patterns/Locale/Locale.vue`? This will allow users of the library to simply drop the Locale component into the codebase and not have to set up LocalStorage fetching in their own `i18n.js` file.

Yes sure, I'll move the changes to the requested file.

File: src/i18n.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ function loadLocaleMessages () {
   return messages
 }
 
-export default new VueI18n({
+export default new VueI18n({
```
Comment (dhruvkb) [inline src/i18n.js]: Remove the trailing space.

File: src/patterns/Locale/Locale.vue [hunk 2]
```diff
@@ -78,6 +87,10 @@
           text: locale.nativeName
         }))
       }
+    },
+    created(){
+      this.$i18n.locale = localStorage.getItem('locale') || this.language
     }
+
```
Comment (dhruvkb) [inline src/patterns/Locale/Locale.vue]: We do not leave newlines at the start and the end of code blocks demarcated by `{` and `}`.

File: src/patterns/Locale/Locale.vue [hunk 3]
```diff
@@ -68,8 +69,16 @@
         }
       ]
       return {
+        language: '',
```
Comment (dhruvkb) [inline src/patterns/Locale/Locale.vue]: Let's start with a default of 'en'.

File: src/patterns/Locale/Locale.vue [hunk 4]
```diff
@@ -68,8 +69,16 @@
...
         locales: this.localeList || defaultLocales
       }
+    },
+     methods: {
+      // The below method simply stores the selected locale in the localstorage
```
Comment (dhruvkb) [inline src/patterns/Locale/Locale.vue]: Drop the `simply` imo.

**Comment by dhruvkb** [comment]
Resolved in a884c94.

**Pull request closed without merging**